src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.cpp

```diff
@@ -868,8 +644,34 @@ HOSTFXR_UTILITY::FindDotNetFolders(
     do
     {
         std::wstring folder(data.cFileName);
-        pvFolders->push_back(folder);
+        pvFolders.push_back(folder);
     } while (FindNextFileW(handle, &data));
 
     FindClose(handle);
 }
+
+std::wstring
+HOSTFXR_UTILITY::ExpandEnvironmentVariables(const std::wstring & str)
+{
+    DWORD requestedSize = ExpandEnvironmentStringsW(str.c_str(), nullptr, 0);
```

Comment:
Add a comment saying ExpandEnvironmentVariables already includes the null character for the requestedSize.